pkgs/applications/audio/open-music-kontrollers/synthpod.nix

```
@@ -0,0 +1,13 @@
+{ callPackage, lilv, libjack2, alsa-lib, zita-alsa-pcmi, libxcb, xcbutilxrm, sratom, gtk2, gtk3, qt4, qt5,  libvterm-neovim, robodoc, cmake,... } @ args:
+
+callPackage ./generic.nix (args // rec {
+  pname = "synthpod";
+  version = "unstable-2021-10-22";
+
+  url = "https://git.open-music-kontrollers.ch/lv2/synthpod/snapshot/synthpod-6f284bdad882037a522c120af92b96d8abf2de60.tar.xz";
+  sha256 = "sha256-59WBlOKum5Pcmq2CfFfRHCNEa8uPCoBk0kSjFlIcypw=";
+
+  additionalBuildInputs = [ lilv libjack2 alsa-lib zita-alsa-pcmi libxcb xcbutilxrm sratom gtk2 gtk3 qt4 qt5.qtbase qt5.wrapQtAppsHook  libvterm-neovim robodoc cmake ];
```

```suggestion
  additionalBuildInputs = [ lilv libjack2 alsa-lib zita-alsa-pcmi libxcb xcbutilxrm sratom gtk2 gtk3 qt4 qt5.qtbase qt5.wrapQtAppsHook libvterm-neovim robodoc cmake ];
```
You are sure about gtk2 and gtk3 and gt4 and qt5?

---

Yup: https://github.com/OpenMusicKontrollers/synthpod#optional-dependencies-for-plugin-uis

---

You are sure they are actually used and the doc is not just out of date or lists potentials? I am not sure if mixing those works great in nixpkgs.

---

Why do you think the docs are out of date? I'm happy to ask upstream, but would like to understand  why I should be doubting them before I do.
What do you mean by "lists potentials"? Did you perhaps not see these are optional and are now wondering if they are, or did I misinterpret the question?
I know too little to understand why I shouldn't mix these. There are other nix pkgs that have multiple gui toolkits in their dependencies, no?

---

To my understanding plugins can use something of that list.
You shouldn't really mix gtk2, gtk3 or gtk4 and qt4, qt5 or qt6 because they are the same with in different versions. Sure you can mix gtk3 and qt5 but mixing gtk2 and gtk3 sounds like trouble to me.